core/scheduler/handler/v1beta1/job_run.go

```diff
@@ -0,0 +1,190 @@
+	GetJobRuns(ctx context.Context, projectName tenant.ProjectName, jobName scheduler.JobName, criteria *scheduler.JobRunsCriteria) ([]*scheduler.JobRunStatus, error)
+	UploadToScheduler(ctx context.Context, projectName tenant.ProjectName, namespaceName string) error
+}
+
+type Notifier interface {
+	Push(ctx context.Context, event scheduler.Event) error
+}
+
+type JobRunHandler struct {
+	l        log.Logger
+	service  JobRunService
+	notifier Notifier
+
+	pb.UnimplementedJobRunServiceServer
+}
+
+func (h JobRunHandler) JobRunInput(ctx context.Context, req *pb.JobRunInputRequest) (*pb.JobRunInputResponse, error) {
+	projectName, err := tenant.ProjectNameFrom(req.GetProjectName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run input for "+req.GetJobName())
+	}
+
+	jobName, err := scheduler.JobNameFrom(req.GetJobName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run input for "+req.GetJobName())
+	}
+
+	executor, err := scheduler.ExecutorFromEnum(req.InstanceName, req.InstanceType.String())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run input for "+req.GetJobName())
+	}
+
+	err = req.ScheduledAt.CheckValid()
+	if err != nil {
+		return nil, errors.GRPCErr(errors.InvalidArgument(scheduler.EntityJobRun, "invalid scheduled_at"), "unable to get job run input for "+req.GetJobName())
+	}
+
+	runConfig, err := scheduler.RunConfigFrom(executor, req.ScheduledAt.AsTime(), req.JobrunId)
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run input for "+req.GetJobName())
+	}
+
+	input, err := h.service.JobRunInput(ctx, projectName, jobName, runConfig)
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run input for "+req.GetJobName())
+	}
+
+	return &pb.JobRunInputResponse{
+		Envs:    input.Configs,
+		Files:   input.Files,
+		Secrets: input.Secrets,
+	}, nil
+}
+
+// JobRun currently gets the job runs from scheduler based on the criteria
+// TODO: later should collect the job runs from optimus
+func (h JobRunHandler) JobRun(ctx context.Context, req *pb.JobRunRequest) (*pb.JobRunResponse, error) {
+	projectName, err := tenant.ProjectNameFrom(req.GetProjectName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run for "+req.GetJobName())
+	}
+
+	jobName, err := scheduler.JobNameFrom(req.GetJobName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run for "+req.GetJobName())
+	}
+
+	criteria, err := buildCriteriaForJobRun(req)
+	if err != nil {
+		return nil, errors.GRPCErr(err, "unable to get job run for "+req.GetJobName())
+	}
+
+	var jobRuns []*scheduler.JobRunStatus
+	jobRuns, err = h.service.GetJobRuns(ctx, projectName, jobName, criteria)
+	if err != nil {
+		if errors.IsErrorType(err, errors.ErrNotFound) {
+			return &pb.JobRunResponse{}, nil
+		}
+		return nil, errors.GRPCErr(err, "unable to get job run for "+req.GetJobName())
+	}
+
```

Comment:
@sravankorumilli the old code has a check here 
```
	if len(jobRuns) == 0 {
		return nil, status.Errorf(codes.NotFound, "%s\n job runs not found ",
			req.GetJobName())
	}
```

which we believe is inappropriate, because if expected runs is 0 then this should not be stoping the sensor